crypto/provider_core.c

```diff
@@ -1066,7 +1056,7 @@ static int provider_deactivate(OSSL_PROVIDER *prov, int upcalls,
         return -1;
     }
 
-    CRYPTO_atomic_add(&prov->activatecnt, -1, &count, prov->refcnt_lock);
+    CRYPTO_DOWN_REF(&prov->activatecnt, &count);
```

Comment:
In a previous PR we initialy had activatecnt as a refcount, but then deliberately changed it to an int and used `CRYPTO_atomic_add` for reasons described by @kroeckx. Here you are converting it to a ref count again. This doesn't look right. We might need to do the same work for `CRYPTO_atomic_add()` et al, that you are doing here for ref counts. But probably not in this PR. I'd leave refcnt_lock alone (maybe rename it), so that it can be used for the activatecnt and deal with that one in a subsequent PR which converts atomic ints into a structure.

Reply:
Fixed.